stl/inc/random

```diff
@@ -4571,20 +4575,15 @@ public:
         param_type(initializer_list<_Ty> _Ilist, _Fn _Func) : _Mypbase(_Noinit) {
             if (2 <= _Ilist.size()) {
                 _Bvec.assign(_Ilist);
+
+                for (size_t _Idx = 0; _Idx < _Bvec.size() - 1; ++_Idx) {

```

Comment:
There are several more occurrences in `discrete_distribution`, `piecewise_constant_distribution` and `piecewise_linear_distribution` where we could have `reserve`d before populating `vector`s. Perhaps for another "performance" PR?

Reply:
A separate performance PR would make sense. Cautionary note: a single `reserve` after `vector` construction is fine, but we should be careful to avoid `reserve` in anything that the user can call in a loop for a single `vector`, as that can trigger quadratic complexity. I'm specifically thinking about `dist.param(const param_type&)` here.

Reply:
A pattern like this
https://github.com/microsoft/STL/blob/1fef5f71e5c7807ad966624ef421d6ad97f746a3/stl/inc/random#L247-L251
is ideally replaced with range-based algorithm, which would result in transformed iterators, and the vector would reserve automatically at least for random-access iterator.
Maybe just create issue for now and wait for ranges?

Reply:
This needs to work in C++14 mode and thus will not have ranges avaialble.